scripts/fourmolize.sh

```diff
@@ -5,19 +5,18 @@ set -euo pipefail
 if [[ $# -gt 0 ]]; then
   case "$1" in
     --changes)
-      files=$(git diff --diff-filter=MA --name-only origin/master HEAD -- '*.hs')
-      if [[ -n "$files" ]]; then
-        # Run fourmolu on changes compared to `master`.
-        fourmolu -m inplace $(echo "$files" | grep -v Setup.hs)
-      fi
+      # Run fourmolu on changes compared to `master`.
+      git diff --diff-filter=MA --name-only origin/master HEAD -- '*.hs'
       ;;
     *)
       echo "Invalid option: $1" >&2
       exit 1
       ;;
   esac
 else
-  fourmolu -m inplace $(git ls-files -- '*.hs' | grep -v Setup.hs)
-fi
+  git ls-files -- '*.hs'
+fi \
+  | { grep -v Setup.hs || true; } \
+  | xargs -r fourmolu -m inplace
```

Comment:
shellcheck might have reduced duplication here, but I am not so confident that it resulted in a more readable script. In any case, it is bash, so it was ugly to begin with, therefore it doesn't hurt to make it a bit uglier :smile:

Reply:
It was mostly me rather than `shellcheck`, although it did need some rewriting to get around the issues that `shellcheck` identified.

It's a pity that the `|| true` is needed, but that's a consequence of using `set -o pipefail`. Although that was set before, the grep failures (when there are no non-`Setup,hs` files) were swallowed by being in a variable assignment.